src/hotspot/share/cds/aotClassLocation.cpp

```diff
@@ -903,7 +907,7 @@ void AOTClassLocationConfig::print_dumptime_classpath(LogStream& ls, int index_s
     const char* path = cs->path();
     if (!cs->from_cpattr()) {
       ls.print("%s", sep);
-      if (do_substitute) {
+      if (do_substitute && remove_prefix_len > 0) {
```

Comment:
I had expected this would no longer be necessary once the cp parsing was fixed. ??

Reply:
I removed this change.